packages/amplify-e2e-core/src/utils/rds.ts

```diff
@@ -309,23 +387,19 @@ export const setupRDSInstanceAndData = async (
  */
 
 export const setupRDSClusterAndData = async (config: RDSConfig, queries?: string[]): Promise<ClusterInfo> => {
-  console.log(`Creating RDS ${config.engine} DB cluster with identifier ${config.identifier}`);
-
   const dbCluster = await createRDSCluster(config);
+  console.log(`Creating RDS ${config.engine} DB cluster with identifier ${config.identifier}`);
 
   if (!dbCluster.secretArn) {
     throw new Error('Failed to store db connection config in secrets manager');
   }
 
   const client = new RDSDataClient({ region: config.region });
 
-  // create a new test database with given name
-  const sanitizedDbName = config.dbname.replace(/[^a-zA-Z0-9_]/g, '');
-
   const createDBInput: ExecuteStatementCommandInput = {
     resourceArn: dbCluster.clusterArn,
     secretArn: dbCluster.secretArn,
-    sql: `create database ${sanitizedDbName}`,
+    sql: `create database ${config.dbname}`,
```

Comment:
nit: add a note here that is already sanitized when we generate the random database name.

Reply:
As noted above, this isn't sufficient, since we export the "create database" method so that other callers use it, and there's nothing enforcing those callers to sanitize the name. We need to be in the habit of sanitizing input at the point of usage, even if we believe it's sanitized elsewhere.